### PR TITLE
New JsRT API to parse script as library (internal) code. Scripts marked as libraryCode are internal and hidden from debugger

### PR DESCRIPTION
**Pull request by agarwal-sandeep**
This API can be used to hide host code written in JavaScript

### CODE REVIEW

**Comment by agarwal-sandeep** [comment]
@liminzhu @Yongqu 
Limin and Yong please take a look

File: lib/jsrt/chakracommon.h [hunk 1]
```diff
@@ -813,6 +813,44 @@
             _Out_ JsValueRef *result);
 
     /// <summary>
+    ///     Attribute mask for JsParseScriptWithFlags
+    /// </summary>
+    typedef enum _JsParseScriptAttributes {
```
Comment (Yongqu) [inline lib/jsrt/chakracommon.h]: we usually put the definitions higher on the top of the file

File: lib/jsrt/JsrtCommonExports.inc [hunk 2]
```diff
@@ -90,6 +90,7 @@
     JsParseSerializedScriptWithCallback
     JsRunSerializedScriptWithCallback
     JsParseScript
+    JsParseScriptWithFlags
```
Comment (Yongqu) [inline lib/jsrt/JsrtCommonExports.inc]: @liminzhu can comment on the publishing part.

File: lib/Runtime/Base/ScriptContext.cpp [hunk 3]
```diff
@@ -1753,6 +1759,12 @@ namespace Js
                 grfscr |= (fscrNoAsmJs | fscrNoPreJit);
             }
 
+            if (isLibraryCode)
+            {
+                grfscr |= fscrIsLibraryCode;
+                (*ppSourceInfo)->SetIsLibraryCode();
```
Comment (Yongqu) [inline lib/Runtime/Base/ScriptContext.cpp]: we are not really using utf8 code path now though I'm adding usage for it later. here *ppSourceInfo is not created yet.

File: lib/jsrt/JsrtCommonExports.inc (hunk 2 above)
Comment (liminzhu) [inline lib/jsrt/JsrtCommonExports.inc]: Had a off-line conversation with @agarwal-sandeep yesterday. LGTM.

**Comment by Yongqu** [comment]
LGTM

**Pull request merged**